pkg/lint/rules/chartfile.go

```diff
@@ -104,9 +105,14 @@ func validateChartYamlFormat(chartFileError error) error {
 }
 
 func validateChartName(cf *chart.Metadata) error {
+	var validNameRegex = regexp.MustCompile(`^[a-z0-9](([-a-z0-9]*)?)$`)
+
 	if cf.Name == "" {
```

Comment:
strictly, this check is redundant. But I think the explicit check for an empty name is worthwhile in terms of the messages presented to the user (ie. keep this)